components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIAdminImplTest.java

```diff
@@ -164,4 +171,25 @@ public void getTenantConfigSchema() throws Exception {
         PowerMockito.when(APIUtil.class, "retrieveTenantConfigJsonSchema").thenReturn(schema);
         Assert.assertEquals(apiAdmin.getTenantConfigSchema("abc.com"), schema.toString());
     }
+
+    @Test
+    public void getPolicyByNameAndType() throws Exception {
+        APIPolicy apiPolicy = Mockito.mock(APIPolicy.class);
+        APIManagerConfigurationService apiManagerConfigurationService = Mockito.mock(APIManagerConfigurationService.class);
+        APIManagerConfiguration apiManagerConfiguration = Mockito.mock(APIManagerConfiguration.class);
+        ThrottleProperties throttleProperties = Mockito.mock(ThrottleProperties.class);
+        APIAdmin apiAdmin = new APIAdminImpl();
+        Mockito.when(apiMgtDAO.getAPIPolicy(ArgumentMatchers.anyString(), ArgumentMatchers.anyInt())).thenReturn(apiPolicy);
+        Mockito.when(serviceReferenceHolder.getAPIManagerConfigurationService()).thenReturn(apiManagerConfigurationService);
+        Mockito.when(apiManagerConfigurationService.getAPIManagerConfiguration()).thenReturn(apiManagerConfiguration);
+        Mockito.when(apiManagerConfiguration.getThrottleProperties()).thenReturn(throttleProperties);
+        Mockito.when(throttleProperties.isEnableUnlimitedTier()).thenReturn(true);
+
+        try {
+            Policy policy = apiAdmin.getPolicyByNameAndType(ArgumentMatchers.anyInt(), "api", ArgumentMatchers.anyString());
+            Assert.assertNotNull(policy);
+        } catch (APIManagementException ex) {
+            Assert.fail(ex.getMessage());
+        }
+    }
 }
```

Comment:
The EOF line is removed here. Shall we add that?

Reply:
Fixed by [a07d183](https://github.com/wso2/carbon-apimgt/pull/11386/commits/a07d183069d15b4e30dc01efe8be721d305e71b5)